doc/manual/src/installation/installing-binary.md

```diff
@@ -95,15 +97,87 @@ sudo systemctl stop nix-daemon.service
 sudo systemctl disable nix-daemon.socket
 sudo systemctl disable nix-daemon.service
 sudo systemctl daemon-reload
-
-# If you are on macOS, you will need to run:
-sudo launchctl unload /Library/LaunchDaemons/org.nixos.nix-daemon.plist
-sudo rm /Library/LaunchDaemons/org.nixos.nix-daemon.plist
 ```
 
 There may also be references to Nix in `/etc/profile`, `/etc/bashrc`,
 and `/etc/zshrc` which you may remove.
 
+### macOS
+
+1. Edit `/etc/zshrc` and `/etc/bashrc` to remove the lines sourcing
+   `nix-daemon.sh`, which should look like this:
+
+   ```bash
+   # Nix
+   if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
+     . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
+   fi
+   # End Nix
+   ```
+
+   If these files haven't been altered since installing Nix you can simply put
+   the backups back in place:
+
+   ```console
+   sudo mv /etc/zshrc.backup-before-nix /etc/zshrc
+   sudo mv /etc/bashrc.backup-before-nix /etc/bashrc
+   ```
+
+   This will stop shells from sourcing the file and bringing everything you
+   installed using Nix in scope.
+
+2. Stop and remove the Nix daemon services:
+
+   ```console
+   sudo launchctl unload /Library/LaunchDaemons/org.nixos.nix-daemon.plist
+   sudo rm /Library/LaunchDaemons/org.nixos.nix-daemon.plist
+   sudo launchctl unload /Library/LaunchDaemons/org.nixos.activate-system.plist
+   sudo rm /Library/LaunchDaemons/org.nixos.activate-system.plist
```

Comment:
This is the nix-darwin service, yeah? did you mean to use that one instead of the mount daemon? 

(I was reading over all of this and thinking maybe we could get away with adding a few qualifiers to a few of these steps to say something like "You may not have X if you if you are running macOS Mojave (or earlier), or if you installed Nix before Y.")

Reply:
Ah, you're right of course. I was originally writing the instructions from a "I want to get rid of all things Nix," point of view but this would be more suited to the nix-darwin manual.

Maybe we can use the fstab step as the first differentiating step and add a note along the lines of "If you don't have this entry you probably have an older install on a pre-Catalina macOS, you won't have to do steps X, Y and Z and you should `rm /nix` instead?"

Reply:
Ah, yes, similar to what you say in the very next comment but just localized to the first step where someone might notice a difference.

Alternatively we could make all the post-Catalina steps a sublist of a single step and then we can say to just skip the one step. Keeping them at the top-level list is slightly more flexible, the current order is based on not ending up with a broken setup at any point and going from easy to un-undo to harder to un-undo. However, I do think we could do all the post-Catalina specific stuff grouped.